src/main.rs

```diff
@@ -1,14 +1,34 @@
 use clap::Parser;
-use std::path::PathBuf;
 
-pub type Result<T, E = Box<dyn std::error::Error + Send + Sync + 'static>> =
-    core::result::Result<T, E>;
+mod errors;
+mod grow;
+mod identify;
+mod lsblk;
+
+use crate::errors::Result;
+use crate::identify::zfs_find_partitions_in_pool;
 
 #[derive(Debug, Parser)]
 #[clap(about, long_about = None)]
 struct Options {
     /// Specified ZPool name to lookup in ZFS
     zpool_name: String,
+
+    /// Automatically grow all candidate partitions (true|false)
+    #[clap(long, parse(try_from_str = true_or_false), default_value_t)]
+    automatically_grow: bool,
+
+    /// Don't make any changes (true|false)
+    #[clap(long, parse(try_from_str = true_or_false), default_value_t)]
+    dry_run: bool,
```

Comment:
What's the reasoning for having these arguments take a value instead of being based on the presence of the flag/switch?